crates/bevy_core/src/name.rs

```diff
@@ -8,6 +8,7 @@ use std::{
 };
 
 /// Component used to identify an entity. Stores a hash for faster comparisons
```

Comment:
I think this needs a comment explaining that names are not guaranteed to be globally unique. And probably also that `Entity` should be used as the default "unique identifier" instead.

Especially with `Hash` being implemented, that's an easy and painful mistake to make.